pkg/healthcheck/healthcheck.go

```diff
@@ -213,67 +287,95 @@ func (hc *HealthChecker) addKubernetesAPIChecks() {
 			return
 		},
 	})
+}
 
-	if hc.ShouldCheckKubeVersion {
-		hc.checkers = append(hc.checkers, &checker{
-			category:    KubernetesAPICategory,
-			description: "is running the minimum Kubernetes API version",
-			check: func() error {
-				return hc.kubeAPI.CheckVersion(hc.kubeVersion)
-			},
-		})
-	}
+func (hc *HealthChecker) addKubernetesVersionChecks() {
+	hc.checkers = append(hc.checkers, &checker{
+		category:    KubernetesVersionCategory,
+		description: "is running the minimum Kubernetes API version",
+		check: func() error {
+			return hc.kubeAPI.CheckVersion(hc.kubeVersion)
+		},
+	})
 }
 
-func (hc *HealthChecker) addLinkerdPreInstallChecks() {
+func (hc *HealthChecker) addLinkerdPreInstallClusterChecks() {
+	hc.checkers = append(hc.checkers, &checker{
+		category:    LinkerdPreInstallClusterCategory,
+		description: "control plane namespace does not already exist",
+		check: func() error {
+			return hc.checkNamespace(hc.ControlPlaneNamespace, false)
+		},
+	})
+
+	hc.checkers = append(hc.checkers, &checker{
+		category:    LinkerdPreInstallClusterCategory,
+		description: "can create Namespaces",
+		check: func() error {
+			return hc.checkCanCreate("", "", "v1", "Namespace")
+		},
+	})
+
+	// TODO: refactor with LinkerdPreInstallSingleNamespaceChecks
 	roleType := "ClusterRole"
 	roleBindingType := "ClusterRoleBinding"
```

Comment:
Now that's you've split the RBAC checks into multiple separate methods, I think it's clearer to just hardcode everything, rather than worrying about code reuse. I'm inclined to just remove these local vars. Something like:

```diff
diff --git a/pkg/healthcheck/healthcheck.go b/pkg/healthcheck/healthcheck.go
index 24b9722e..31c99ce2 100644
--- a/pkg/healthcheck/healthcheck.go
+++ b/pkg/healthcheck/healthcheck.go
@@ -316,23 +316,19 @@ func (hc *HealthChecker) addLinkerdPreInstallClusterChecks() {
 		},
 	})
 
-	// TODO: refactor with LinkerdPreInstallSingleNamespaceChecks
-	roleType := "ClusterRole"
-	roleBindingType := "ClusterRoleBinding"
-
 	hc.checkers = append(hc.checkers, &checker{
 		category:    LinkerdPreInstallClusterCategory,
-		description: fmt.Sprintf("can create %ss", roleType),
+		description: "can create ClusterRoles",
 		check: func() error {
-			return hc.checkCanCreate("", "rbac.authorization.k8s.io", "v1beta1", roleType)
+			return hc.checkCanCreate("", "rbac.authorization.k8s.io", "v1beta1", "ClusterRole")
 		},
 	})
 
 	hc.checkers = append(hc.checkers, &checker{
 		category:    LinkerdPreInstallClusterCategory,
-		description: fmt.Sprintf("can create %ss", roleBindingType),
+		description: "can create ClusterRoleBindings",
 		check: func() error {
-			return hc.checkCanCreate("", "rbac.authorization.k8s.io", "v1beta1", roleBindingType)
+			return hc.checkCanCreate("", "rbac.authorization.k8s.io", "v1beta1", "ClusterRoleBinding")
 		},
 	})
 
```

Same goes for the checks in the `addLinkerdPreInstallSingleNamespaceChecks` func.

Reply:
heh, i did exactly that in the next PR: https://github.com/linkerd/linkerd2/pull/2080/files#diff-d4056ff163bcf2aeacefb2a34164563cR270

Reply:
Awesome, yep, carry on!